File: src/main/java/inno/jago/ast/declaration/FunctionDeclarationNode.java
package inno.jago.ast.declaration;

import inno.jago.ast.StatementNode;
import inno.jago.ast.entity.Entity;
import inno.jago.ast.SignatureNode;
import lombok.Getter;
import lombok.Setter;

/*
functionDecl        :   FUNC functionName signature ( functionBody )?;
 */
@Getter
@Setter
public class FunctionDeclarationNode extends Entity {
    private String functionName;

    private SignatureNode signatureNode;

    private StatementNode functionBody;

    public FunctionDeclarationNode(String functionName, SignatureNode signatureNode, StatementNode functionBody) {
        this.functionName = functionName;
        this.signatureNode = signatureNode;
        this.functionBody = functionBody;
    }
}

File: src/main/kotlin/inno/jago/ast/decl/FunctionDeclarationNode.kt
package inno.jago.ast.decl

import inno.jago.lexer.Pos
import inno.jago.ast.type.TypeNode
import inno.jago.ast.expression.ExpressionNode

class FunctionDeclarationNode (
    pos: Pos,
    functionName : String,
    signature : SignatureNode,
    functionBody : BlockStatementNode
) : TopLevelDeclNode(pos = pos)



